Modified readme to talk about Websockets reborn specifically

relyt29 opened the pull request:
If anyone else has something they think should be added please comment.

Clam- committed:
Extend DPAD support.

Allows the hardware DPAD to control more of the UI.

Closes #4989
// FREEBIE

kaladivo committed:
Add cancel to device management network failure alert dialog

Closes #4844

JavaJens commented on README.md:
```diff
@@ -1,40 +1,37 @@
 ## Contributing Bug reports
 We use GitHub for bug tracking. Please search the existing issues for your bug and create a new one if the issue is not yet tracked!
 
 https://github.com/WhisperSystems/Signal-Android/issues
```
Maybe we should change this also?

Diapolo replied:
I suggest:

<pre>
## Contributing Bug reports related to WebSocket support
We use GitHub for bug tracking. Please search the existing issues for your bug and create a new one if the issue is not yet tracked!
 
https://github.com/JavaJens/TextSecure/issues

## Contributing Bug reports upstream for the base application
They also use GitHub for bug tracking. Please search the existing issues for your bug and create a new one if the issue is not yet tracked!
 
https://github.com/WhisperSystems/Signal-Android/issues
</pre>

JavaJens commented on README.md:
`````diff
@@ -1,40 +1,37 @@
-and run: ````./gradlew tasks installArchives```` to install in local maven directory.
+## THIS CODE IS ALPHA QUALITY AND HAS KNOWN BUGS. PLEASE BE AWARE AND USE AT YOUR OWN RISK.
+
+In specific, Signal ZRTP encrypted calls do not work at this time. That code is commented out of this websocket-only version.
`````
I think this is also an upstream problem right?
From what I understand ZRTP works, but the continuity checks are disabled. That means you have to verify your partner at every call and not only the first.

relyt29 replied:
I have no idea. I haven't tried to do ZRTP at all because I just assumed initially it would not work with GCM. Have you actually tried to call someone? I thought you couldn't receive calls or something similar.

Diapolo commented on README.md:
```diff
@@ -1,40 +1,37 @@
-# Signal Android 
+# Signal Android Websockets-Only
```
I'd like to not call it that way, because of pull https://github.com/JavaJens/TextSecure/pull/40, which I still think should be merged. Also it's written `WebSocket` :).

Why not use `Signal Android with WebSocket support`?

Diapolo commented on README.md:
```diff
@@ -1,40 +1,37 @@
 
 Signal uses your phone's data connection (WiFi/3G/4G) to communicate securely, optionally supports plain SMS/MMS to function as a unified messenger, and can also encrypt the stored messages on your phone.
 
-Currently available on the Play store.
```
We could add the F-Droid repo from @xmikos (https://eutopia.cz/experimental/fdroid/repo/) and mention this as an alternative to self-building. Is there some way to make the app hash comparable so that self-compiled ones and the ones from @xmikos match? To be sure it's not modified in a mallicous way?

JavaJens replied:
I don't think there is such way. Therefore I don't want to endorse this. As @f41c0r mentioned elsewhere, this is development code.

Diapolo commented on README.md:
```diff
@@ -1,40 +1,37 @@
-
 #WebSocket Support#
 
 This branch adds rudimentary WebSocket-ONLY support to Signal Private Messenger.
```
Nit: You could remove the `-ONLY` if we merge https://github.com/JavaJens/TextSecure/pull/40.

Diapolo commented on README.md:
```diff
@@ -1,40 +1,37 @@
-Instructions on how to setup your development environment and build Signal can be found in  [BUILDING.md](https://github.com/WhisperSystems/Signal-Android/blob/master/BUILDING.md).
+General instructions on how to setup your development environment and build vanilla Signal can be found in  [BUILDING.md](https://github.com/WhisperSystems/Signal-Android/blob/master/BUILDING.md).
 
 If you're new to the Signal codebase, we recommend going through our issues and picking out a simple bug to fix (check the "easy" label in our issues) in order to get yourself familiar.
```
We have no "easy" label, so let's remove it.

JavaJens replied:
This links to the main/vanilla repo.

Diapolo replied:
Right, but is that helpful for this fork to link to vanilla here?

relyt29 replied:
I think its of some small use. There are some notes in the BUILDING.md file that are worth reading.

schachmat commented:
Since per my understanding this fork is just for one feature, I would not merge this PR. When merging new upstream changes I always closely check the new diff against upstream and this README change would be very annoying. If you want to have a more accurate readme, please let @xmikos create his own fork for support related questions for LibreSignal and change the readme there as I have argued in #72.

haffenloher committed:
Use high-res callcard for TransparentContactPhoto

Closes #5228

// FREEBIE

moxie0 committed:
Add Dockerfile for an Android build environment

// FREEBIE

passy committed:
Change lock icons

Closes #5325
// FREEBIE

2-4601 committed:
Update issue and pull request templates

Closes #5285
// FREEBIE

SecUpwN committed:
Modified README for LibreSignal

Closes #8

mimi89999 committed:
Minor changes in README.

moxie0 committed:
Drop some packages from spongycastle to accomodate GB dexopt

Fixes #5269
// FREEBIE

moxie0 committed:
Be more persistent about completing the spk job

// FREEBIE

SecUpwN committed:
Updated CONTRIBUTING.md to fit LibreSignal

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.14.0

// FREEBIE

mimi89999 committed:
Merge branch 'master' of https://github.com/WhisperSystems/Signal-Android

moxie0 committed:
Forgot that we do need the JCE EC stuff

Fixes #5337
// FREEBIE

moxie0 committed:
Bump version to 3.14.1

// FREEBIE

moxie0 committed:
Fall back to lollipop MMS API if PDU response isn't success

Fixes #4131
// FREEBIE

passy committed:
Change ic_menu_lock_dark to full opacity white

Fixes #5342
Closes #5343
// FREEBIE

mimi89999 committed:
Merge branch 'master' of https://github.com/WhisperSystems/Signal-Android

moxie0 committed:
Bump version to 3.14.2

// FREEBIE

cascheberg committed:
replace toasts in PassphraseChangeActivity with errors

Closes #5347
// FREEBIE

mimi89999 committed:
Merge branch 'master' of https://github.com/WhisperSystems/Signal-Android

mimi89999 committed:
Change applicationId to "org.libresignal"

Fixes #12

cascheberg committed:
prepare to use Signal in verification SMS text

Closes #5352
// FREEBIE

cascheberg committed:
fix OutgoingRinger crashes

Closes #5357

mimi89999 committed:
Fix F-Droid link.

moxie0 committed:
TextSecure is now SignalService

// FREEBIE

haffenloher committed:
Accept verification SMSes with missing colons

Apparently some Argentinian mobile providers strip the colons from
Signal's verification SMS messages.

Fixes #5363
Fixes #3490
Closes #5371

// FREEBIE

moxie0 committed:
Prohibit sharing file:// Uris

Just in case someone tries something sneaky...

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.15.0

// FREEBIE

mimi89999 committed:
Merge https://github.com/WhisperSystems/Signal-Android

moxie0 committed:
Fix identity key formatting regression

Fixes #5376
// FREEBIE

moxie0 committed:
Bump version to 3.15.1

// FREEBIE

mimi89999 committed:
Fix F-Droid build.

mimi89999 committed:
Merge https://github.com/WhisperSystems/Signal-Android

moxie0 committed:
Check file uri sharing owner rather than prohibiting outright

Fixes #5381
// FREEBIE

moxie0 committed:
Bump version to 3.15.2

// FREEBIE

mimi89999 committed:
ShortcutBadger as a submodule.

mimi89999 committed:
Merge https://github.com/WhisperSystems/Signal-Android

mimi89999 committed:
Remove binary files.

mimi89999 committed:
Ignore .so libs with libs/.gitignore

mimi89999 committed:
Update BUILDING.md

tuxayo committed:
readme: details about WebSocket usage and GCM replacement

Fixes #17
Closes #23

cascheberg committed:
add contextual menu items to NewConversationActivity

Closes #5390
// FREEBIE

cascheberg committed:
improve invite strings

Fixes #4869
Closes #5393
//FREEBIE

Commit:
Ensure the browser does not re-use the current tab by removing the
Browser.EXTRA_APPLICATION_ID Extra instead of adding the newer
EXTRA_CREATE_NEW_TAB extra.

Closes #5397
Fixes #4874
Fixes #5356

2-4601 committed:
Update PR template to help auto-closing issues

Closes #5399

// FREEBIE

moxie0 committed:
Fix APK comparison script.

Fixes #5404

// FREEBIE

mimi89999 committed:
Remove Google Repository from BUILDING.md

Fixes #26

2-4601 committed:
Clarify that first commit matters for keywords

// FREEBIE

cascheberg committed:
fix avatar content type

Closes #5417
// FREEBIE

Ofenhed committed:
Stop crash on multiple attachment selected

Fixed #5272
Closes #5440
// FREEBIE

mimi89999 committed:
Add a link to the FAQ.

mimi89999 committed:
Merge https://github.com/WhisperSystems/Signal-Android

mimi89999 committed:
Update the README.

cascheberg committed:
don't play in-thread notification sound in priority mode

Fixes #3337
Closes #5439

// FREEBIE

FeuRenard committed:
Calculate SMS characters respecting encoding

Fixes #1171
Closes #5448

// FREEBIE

Commit:
Allow choosing transmitted contacts

mimi89999 committed:
Remove "BitHub reward" and WS CLA from PR template

cascheberg committed:
prevent contact number NPE

Closes #5460
Fixes #5454
//FREEBIE

mimi89999 committed:
Merge pull request #35 from PrivacyPlus/choose-transmitted-contacts

Choose transmitted contacts

jonathancross committed:
Bitly links now support HTTPS - Yay!

Both invite links now work over `https` -- yes I tested  :-)
More info:

https://github.com/EFForg/https-everywhere/pull/4505
http://webmasters.stackexchange.com/questions/87163/do-bitly-branded-short-domains-support-https/92407#92407

PS: https://sgnl.link/1IvurmD also works.

Closes #5460
Fixes #4060

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Fix broken escapes

// FREEBIE

moxie0 committed:
Bump version to 3.16.0

// FREEBIE

cascheberg committed:
Resize fast scroller when keyboard is shown

Fixes #5487
Closes #5505
// FREEBIE

mimi89999 committed:
Update README.md

nrizzio committed:
Correct a couple more invite shortlinks to use HTTPS

Closes #5511
// FREEBIE

Commit:
Fixing merge BS from pulling in LibreSignal changes

mimi89999 committed:
Merge https://github.com/WhisperSystems/Signal-Android

mimi89999 committed:
Fix my merge mistake in res/values/strings.xml

Commit:
Workaround to add tsdevice:/? string when QR codes won't work

Commit:
Merge https://github.com/LibreSignal/LibreSignal master into f41c0r fork

eighthave committed:
respond to generic panic trigger Intent by locking

PanicKit provides a common framework for creating "panic button"
apps that can trigger actions in "panic responder" apps.  In this
case, the response is to lock the app, if it has been configured
to do so.

As previously discussed in #5341

Closes #5550

//FREEBIE

cascheberg committed:
mark group update messages as unread

Closes #5575
// FREEBIE

cascheberg committed:
ignore fast sender clock

Closes #5577
// FREEBIE

FeuRenard committed:
Respect privacy settings on ticker content for Android <5

Fixes #5517
Closes #5592

// FREEBIE

FeuRenard committed:
Resize group images

Fixes #5461
Closes #5597

// FREEBIE

FeuRenard committed:
Use Signal's language for media preview timestamp

Fixes #3449
Closes #5600

// FREEBIE

jollheef committed:
Use gravity instead of calculations

For avoid freezes when trying to add an attachment

Fixes #5583
Closes #5601
// FREEBIE

moxie0 committed:
Bump version to 3.16.1

// FREEBIE

cascheberg committed:
fix ConversationItem recipients listener

Fixes #4420
Closes #5615
// FREEBIE

FeuRenard committed:
Catch ActivityNotFoundExceptions

Fixes #3726
Closes #5617
// FREEBIE

moxie0 committed:
Support for disappearing messages

// FREEBIE

FeuRenard committed:
Mark verified messages as push

Fixes #4769
Closes #5627
// FREEBIE

moxie0 committed:
Bump version to 3.17.0

// FREEBIE

moxie0 committed:
Support for updated fingerprint format

// FREEBIE

moxie0 committed:
Support for syncing contact colors and block lists

Closes #5638
// FREEBIE

FeuRenard committed:
Clear invite reminder if recipient isn't a contact

Fixes #5637
Closes #5640
// FREEBIE

moxie0 committed:
Hide disappearing messages menu for coordinated release

// FREEBIE

moxie0 committed:
Make fingerprint changes optionally non-blocking

Also complete the rename from "identity" to "safety numbers."

// FREEBIE

FeuRenard committed:
Preserve maxLength for EmojiEditText

Addresses #3388
Closes #5649
// FREEBIE

cascheberg committed:
always set LED notification

Closes #5654
// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Fix translation

// FREEBIE

moxie0 committed:
Bump version to 3.18.0

// FREEBIE

moxie0 committed:
Add EXPIRES_IN column to matrix cursor archived placeholder

Fixes #5656

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.18.1

// FREEBIE

FeuRenard committed:
Use contact color in VerifyIdentityActivity

Fixes #3630
Closes #5664
// FREEBIE

FeuRenard committed:
Let LogSubmitActivity adapt to theme

Addresses #2625
Closes #5665
// FREEBIE

FeuRenard committed:
Theme number picker text (disappearing messages)

Fixes #5659
Closes #5667
// FREEBIE

FeuRenard committed:
Allow 2nd person conjugation in timer messages

Fixes #5658
// FREEBIE

Commit:
Merge upstream master WhisperSystems/Signal-Android in feature/websocket-reborn

Manually fixed bunches of files that didnt merge smoothly.

moxie0 committed:
Don't display save for conversation actions

// FREEBIE
Fixes #5661
Closes #5668

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.18.2

// FREEBIE

cascheberg committed:
Fix scrollbars

Fixes #5487
Closes #5671

// FREEBIE

moxie0 committed:
Fix divide by zero error

Fixes #5674

// FREEBIE

moxie0 committed:
Bump version to 3.18.3

// FREEBIE

cascheberg committed:
Fix display of secure session reset records

Fixes #5605
Closes #5680

// FREEBIE

moxie0 committed:
Drop malformed RTP packets

// FREEBIE

moxie0 committed:
Upgrade libsignal-service

// FREEBIE

moxie0 committed:
Add identity changed notification even in blocking mode

// FREEBIE

FeuRenard committed:
Implement onStartLoading() in DeviceListLoader

Fixes #4641
Closes #5682
// FREEBIE

snaggen committed:
Initial Android Auto support

This adds android auto support accordign to
https://developer.android.com/training/auto/messaging/index.html#messaging
However, since android auto is not officially supported in my country,
the functionality is limited. Which means that I have not been able
to fully test everything yet.

What work is:
* Message notification is shown.
* When you click on it, the message is read.

Closes #5880

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.19.0

// FREEBIE

Commit:
Merge branch 'master' of github:WhisperSystems/Signal-Android into feature/websocket-reborn

Merge 2 important security fixes - see libtextsecure-java-jank commit log for deets

RiseT committed:
String improvement: Blocking and unblocking contacts

This fixes/improves several strings related to blocking and unblocking
contacts regarding:

- Wording consistency
- Compliance with Google's Design Guidelines
- Clarity

Closes #5696
// FREEBIE

RiseT committed:
Fix string: Adding group members

This changes the string "Add member" to "Add members" as the
corresponding list is multi-selectable.

Closes #5697
// FREEBIE

moxie0 committed:
Fix contact update NPE

Fixes #5704
// FREEBIE

moxie0 committed:
Bump version to 3.19.1

// FREEBIE

FeuRenard committed:
Add gallery button to MediaPreview

// FREEBIE

Commit:
Added Samsung multi window support

Closes #5714
//FREEBIE

Commit:
Merge upstream 3.19.1 into Signal-Jank

moxie0 committed:
Canonicalize numbers in fingerprint scan

Closes #5693
Closes #5646

// FREEBIE

moxie0 committed:
Safety number string fixes

// FREEBIE

moxie0 committed:
Enable disappearing message controls

// FREEBIE

brinlyau committed:
Fix typo that breaks build.

// FREEBIE

cascheberg committed:
Fix timer update group notification expiry

Closes #5727
// FREEBIE

moxie0 committed:
Start disappearing message timer on read sync messages

Fixes #5733
// FREEBIE

moxie0 committed:
Fix for expiration update sync sent messages

// FREEBIE

moxie0 committed:
Disable non-blocking identity changes for iOS compat

// FREEBIE

moxie0 committed:
Update language translations

// FREEBIE

moxie0 committed:
Remove missing string

// FREEBIE

moxie0 committed:
Update Dockerfile for new build tools

// FREEBIE

moxie0 committed:
Bump version to 3.20.0

// FREEBIE

moxie0 committed:
Increase minLines on disappearing messages duration dialog

This isn't foolproof across all translations, but it's probably
the best we can do.

Fixes #5746

// FREEBIE

moxie0 committed:
Fix for fragment transaction crash in verify identity activity

Fixes #5747

// FREEBIE

moxie0 committed:
Upgrade gradle tools version

// FREEBIE

moxie0 committed:
Don't transmit SMS contact information to multi-device on change

Fixes #5710

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Update fingerprint view on main thread

// FREEBIE

moxie0 committed:
Revert build tools upgrade, seems to negatively affect reproducibility

// FREEBIE

moxie0 committed:
Bump version to 3.20.1

// FREEBIE

RiseT committed:
Make 'Disappears' in message details translatable

Fixes #5755
Closes #5756

// FREEBIE

moxie0 committed:
Fix bad projection on MMS sync read query

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.20.2

// FREEBIE

RiseT committed:
Change DateUtils string 'Now' to 'Just now'

Changes the DateUtils string 'Now' to 'Just now' which is a better fit
imho.
Used for messages in the conversation view that e. g. just have been
received.

// FREEBIE

Closes #5761

moxie0 committed:
Don't crash on recipient preference change for non-signal user

Fixes #5760
// FREEBIe

moxie0 committed:
Update language translations

// FREEBIE

moxie0 committed:
Bump version to 3.20.3

// FREEBIE

FeuRenard committed:
Make 'Tap to scan' readable in dark theme

Fixes #5749
Closes #5762
//FREEBIE

haffenloher committed:
Increase JPEG quality & slow down quality decrease

Related #672
Closes #5772

//FREEBIE

moxie0 committed:
Don't mark expiration updates as unread

Fixes #5768
// FREEBIE

RiseT committed:
Make message and thread records for changed safety numbers translatable

Fixes #5769
Closes #5771

Makes the message and thread records for 'Your safety numbers with %s
have changed' translatable.

// FREEBIE

moxie0 committed:
Start expiration timer when marked read from notification

// FREEBIE

moxie0 committed:
Update language translations

// FREEBIE

moxie0 committed:
Bump version to 3.20.4

// FREEBIE

zaprice committed:
Do not disable multiline input when enter key sends

Fixes #5264
Closes #5783

moxie0 committed:
Giphy integration

// FREEBIE

Commit:
Add shadow under compose view on scroll

Fixes #5098
Closes #5796
// FREEBIE

Oishikatta committed:
Improve #3990

By catching the crash Signal remains unlocked. The user can quickly
retry sharing from the previous app and it will work, rather than
having to return to the home screen and unlock Signal.

Closes #5805

// FREEBIE

moxie0 committed:
Only load stills for low memory devices

// FREEBIE

moxie0 committed:
Limit giphy search bar to one line

// FREEBIE

moxie0 committed:
Bump up push media max edge to 2048 and max size to 4MB

Fixes #672
// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.21.0

// FREEBIE

moxie0 committed:
Close giphy keyboard when search button is pressed

Fixes #5807
// FREEBIE

RiseT committed:
"No results found." --> "Nothing found"

Closes #5814

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.21.1

// FREEBIE

moxie0 committed:
Use smaller gifs to improve load time

// FREEBIE

moxie0 committed:
Missed some glide caching

Fixes #5817
// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.21.2

// FREEBIE

moxie0 committed:
Fix NPE

Fixes #5829
// FREEBIE

moxie0 committed:
Better null result handling

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.21.3

// FREEBIE

amire80 committed:
Change "GiF" to "GIF" in an English user interface string

Closes #5830

FeuRenard committed:
Add forward button to MediaPreview

Closes #5833
Related to #4906
// FREEBIE

nikwen committed:
Get the automated tests up and running again

* Fix the build.gradle file
* Use the correct VisibleForTesting class
* Fix the Android test setUp() logic
* Enable a test in DeliveryReceiptJobTest.java where the @Test anotation was missing

It should be noted that the tests in AttachmentDatabaseTest.java fail.

Fixes #3474
Closes #5882

FeuRenard committed:
Lower memory consumption of Giphy browser

Fixes #5831
// FREEBIE

FeuRenard committed:
Recycle correctly instead of turning it off

Closes #5841
Fixes #5831

moxie0 committed:
Enable non-blocking safety number changes

// FREEBIE

FeuRenard committed:
Add share button for safety numbers

Fixes #5798
Closes #5854
// FREEBIE

moxie0 committed:
Update libsignal-service to 2.4.0

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Update dependency

// FREEBIE

moxie0 committed:
Bump version to 3.22.0

// FREEBIE

moxie0 committed:
Make blocking SN changes the default for now

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.22.1

// FREEBIE

moxie0 committed:
Add copy and compare safety numbers from clipboard actions

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.22.2

// FREEBIE

FeuRenard committed:
Hide linked devices pref for unregistered users

Fixes #4860
Closes #5872
// FREEBIE

moxie0 committed:
Support recovering forgotten/unknown group info from sender

Closes #5876
// FREEBIE

moxie0 committed:
If Signal isn't enabled for receiving SMS, don't allow sending SMS

Closes #5877
// FREEBIE

moxie0 committed:
Improve animations on audio controls, enable push to talk

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Squelch notifications for active desktop converastions

Additionally, limit audible notifications to once every 2 seconds.

// FREEBIE

Fixes #4905
Fixes #3165
Closes #5813

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.23.0

// FREEBIE

moxie0 committed:
Support for "recent photos" thumbnailsabove attachment selector

// FREEBIE

Closes #5893

moxie0 committed:
Support for playing video in-app

// FREEBIE

Closes #5892

RiseT committed:
Change safety number strings to be singular

// FREEBIE

Closes #5894

moxie0 committed:
Move null recipients check

// FREEBIE

Closes #5800

FeuRenard committed:
Catch up on renaming 'All images' button

Closes #5896
// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Fix broken build (grrr @RiseT)

// FREEBIE

moxie0 committed:
Bump version to 3.24.0

// FREEBIE

nrizzio committed:
Fix the content description of the audio record button

Fixes #5910
Closes #5911
// FREEBIE

RiseT committed:
Make description strings in audio_view.xml translatable.

Closes #5916
// FREEBIE

Commit:
Accessible content labels for attachment icons

Fixes #5908
Closes #5917
// FREEBIE

Trolldemorted committed:
fix NPE in PushGroupUpdateJob

Fixes #5919
Closes #5921
// FREEBIE

moxie0 committed:
Bump version to 3.24.1

// FREEBIE

amire80 committed:
Remove comma splice from an English interface string

moxie0 committed:
Support for stickers and scribbles

// FREEBIE

moxie0 committed:
Updated emoji set

// FREEBIE

moxie0 committed:
A slight foray into the modern era

// FREEBIE

moxie0 committed:
Support for generating video thumbnails

// FREEBIE

moxie0 committed:
Use downsampled GIFs for MMS messages

Fixes #5958
// FREEBIE

moxie0 committed:
Update libsignal-service

Fixes #5928

// FREEBIE

moxie0 committed:
Don't print safety number changes in inactive groups

Fixes #5930
// FREEBIE

Commit:
Merged in updates from upstream, not bothering to support location sending

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Update photoview library

Fixes #5955

// FREEBIE

moxie0 committed:
No need to run this job on install

// FREEBIE

moxie0 committed:
Update Dockerfile to support new compileSdk target

// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.25.0

// FREEBIE

moxie0 committed:
Remove superfluous icon and padding that came with sdk 24

Fixes #5965
// FREEBIE

moxie0 committed:
Scale down bitmap during scribble rendering on lowmem devices

Fixes #5969

// FREEBIE

Commit:
Merged in updates from upstream 2.24

Commit:
Merge in upstream 2.25

moxie0 committed:
Revert support library update, the future wasn't all we hoped for

Fixes #5965
// FREEBIE

moxie0 committed:
Update emoji category images to fix SDK <21 tinting issues

Fixes #5973
// FREEBIE

moxie0 committed:
Updated language translations

// FREEBIE

moxie0 committed:
Bump version to 3.25.1

// FREEBIE

moxie0 committed:
Support for censorship circumvention in Egypt and UAE

// FREEBIE

moxie0 committed:
Bump version to 3.25.2

// FREEBIE

moxie0 committed:
Fix avatar download crash bug

Fixes #5980
// FREEBIE

moxie0 committed:
Bump version to 3.25.3

// FREEBIE

moxie0 committed:
Add Cuba and Oman to domain fronting list

// FREEBIE

Commit:
Merge in commits from upstream. What am I doing with my life?

relyt29 commented:
this has long since become irrelevant
